packages/kit/src/runtime-config.ts

```diff
@@ -0,0 +1,98 @@
+import process from 'node:process'
+import destr from 'destr'
+import { snakeCase } from 'scule'
+import { klona } from 'klona'
+
+import defu from 'defu'
+import { useNuxt } from './context'
+import { useNitro } from './nitro'
+
+/**
+ * Access 'resolved' Nuxt runtime configuration, with values updated from environment.
+ *
+ * This mirrors the runtime behavior of Nitro.
+ */
+export function useRuntimeConfig () {
+  const nuxt = useNuxt()
+  return applyEnv(klona(nuxt.options.nitro.runtimeConfig!), {
+    prefix: 'NITRO_',
+    altPrefix: 'NUXT_',
+    envExpansion: nuxt.options.nitro.experimental?.envExpansion ?? !!process.env.NITRO_ENV_EXPANSION,
+  })
+}
+
+/**
+ * Update Nuxt runtime configuration.
+ */
+export function updateRuntimeConfig (runtimeConfig: Record<string, unknown>) {
+  const nuxt = useNuxt()
+  Object.assign(nuxt.options.nitro.runtimeConfig as Record<string, unknown>, defu(runtimeConfig, nuxt.options.nitro.runtimeConfig))
+
+  try {
+    return useNitro().updateConfig({ runtimeConfig })
+  } catch {
+    // Nitro is not yet initialised - we can safely ignore this error
+  }
+}
+
+/** Internal */
+
+type EnvOptions = {
+  env?: Record<string, unknown>
+  prefix?: string
+  altPrefix?: string
+  envExpansion?: boolean
+}
+
+function getEnv (key: string, opts: EnvOptions) {
+  const env = opts.env ?? process.env
```

Comment:
discussion as https://github.com/unjs/nitro/pull/2404: it is not really reasonable to me that we are accepting an env object that is not system environment variables yet call it env. it really should be called something else or directly augment `runtimeConfig` object with any source that is not system environment variables for purpose of updating it like for testing env.

PS: It seems both getEnv/applyEnv internals are currently not exposed so I appreciate keeping consistency with Nitro and discuss on it later if you believe differently.

Reply:
Yes, this isn't exposed and doesn't allow using anything other than `process.env` in this PR.

Reply:
if the final goal is to reuse it into kit, it won't allow this. the purpose of this util is get a system/runtime environment variable by name.

Reply:
This configuration option isn't exposed in this PR. But as I commented above, it is very useful in a context like `@nuxt/test-utils` to be able to get overrides from a source other than `process`.

Reply:
> This configuration option isn't exposed in this PR

So I guess no harm in not introducing it until waiting for the upstream discussion. 

---

I'm afraid by introducing this change, we **WILL** be diverged from nitro. I also understand your use case and proposed a better alternative (`updateRuntimeConfig()`) to update the runtime config from different sources. changing the definition of environment variables is something I never would opt nitro for (nor wish nuxt does..)

Reply:
Oh, I misunderstood your earlier comment:

> It seems both getEnv/applyEnv internals are currently not exposed so I appreciate keeping consistency with Nitro and discuss on it later if you believe differently.

I read that as you appreciated that I _had_ kept consistency with Nitro and were happy to discuss later on.

Which I appreciated.

But I realise that's not what you meant. Happy to remove the unused option.

In any case it did not diverge from Nitro behaviour, and of course would have switched to Nitro implementation when it was available.